### ISSUE TITLE
Initial Values are not showing up in ReferenceArrayInput on second render. 

### ISSUE THREAD

**Issue by HemanthEverest** [marmelab/react-admin#4203]
**What you were expecting:**
Expect the initialValue to be displayed on every render

**What happened instead:**
initialValue is not showing on second render onwards.

**Steps to reproduce:**
https://codesandbox.io/s/cool-dust-72uir

- Go to the comments menu.
- Click on the create button.
- Observe that the posts are auto-populated. Do not save or enter any data in the form.
- Go to comments from the menu again.
- Click on the create button.
- Observe that the post are not auto-populated.
- Click on the post's input. Do not select any value.
- The initialValue starts to appear.

**Environment**

* React-admin version: 3.1.1
* Last version that did not exhibit the issue (if applicable): 2.*
* React version: 16.8.6
* Browser: Chrome 79.0.3945.88
* Stack trace (in case of a JS error): No Error

**Comment by ThieryMichel** [comment]
Thanks for the issue.

**Comment by fzaninotto** [comment]
Either I don't understand this bug, or it's been fixed in the meantime. 

I don't see any initialValue in your CodeSandbox. I don't see any problem when displaying the CommentCreate page for the second time. 

Can you rephrase and recheck your CodeSandbox?

**Comment by HemanthEverest** [comment]
@fzaninotto The code sandbox was broken since the way we pass defaultValue to `ReferenceArrayInput` was changed in recent times.

I've fixed the code sandbox. Now you can follow the exact same steps mentioned in the description of the issue to reproduce.

Let me know if you need more info.

Regards,

**Comment by fzaninotto** [comment]
In your example, you're using a `<SelectInput>` inside a `<ReferenceArrayInput>`. It does not make sense, you should use a `<SelectArrayInput>`. But I tested with  `<SelectArrayInput>`, and it does show the same bug.

**Comment by fzaninotto** [comment]
The bug comes from the fact that you placed the defaultValue in the SelectInput rather than the ReferenceArrayInput. I understand that you did that because the referenceArrayInput defaultValue was ignored. It is now fixed in #4641.
